index.js

```diff
@@ -93,7 +98,33 @@ const camelCase = (input, options) => {
 		input = preserveCamelCase(input, toLowerCase, toUpperCase);
 	}
 
-	input = input.replace(LEADING_SEPARATORS, '');
+	let leadingSeparators = LEADING_SEPARATORS;
+
+	let regString = '';
+	if (options.preserveCharacters) {
+		const {preserveCharacters} = options;
+		regString = '[';
+		if (!preserveCharacters.includes('-')) {
+			regString += '-';
+		}
+
+		if (!preserveCharacters.includes('_')) {
+			regString += '_';
+		}
+
+		if (!preserveCharacters.includes('.')) {
+			regString += '.';
+		}
+
+		if (!preserveCharacters.includes(' ')) {
+			regString += ' ';
+		}
```

Comment:
This can be done much simpler. There should also be validation for unsupported characters.